CenterForOpenScience/ember-osf-web
# [ENG-91] New home-page hero banner

## Purpose

To add versions A and B of the hero banner to the new home-page.

## Summary of Changes

- Add `add-your-research` component (version A and B)
- Add `discover-research` component (version A and B)
- Add acceptance test for `new-home` page
- Add integration tests for `add-your-research` and `discover-research` components
- Add feature tab to the development footer to switch between feature-flags

**Version A:**
<img width="741" alt="Screen Shot 2019-04-09 at 12 03 47 AM" src="https://user-images.githubusercontent.com/19379783/55772775-b4010400-5a5b-11e9-8911-6e18dd6d6594.png">
<img width="399" alt="Screen Shot 2019-04-09 at 12 04 19 AM" src="https://user-images.githubusercontent.com/19379783/55772798-c9762e00-5a5b-11e9-991b-f7b221d0a286.png">


**Version B:**
<img width="742" alt="Screen Shot 2019-04-09 at 12 03 37 AM" src="https://user-images.githubusercontent.com/19379783/55772770-af3c5000-5a5b-11e9-8176-8a2720c56836.png">
<img width="398" alt="Screen Shot 2019-04-09 at 12 04 39 AM" src="https://user-images.githubusercontent.com/19379783/55772802-cf6c0f00-5a5b-11e9-8803-6c7d687215de.png">


## Side Effects

This will add components for only the new-home page.  It shouldn't affect anything outside of that page.

## Feature Flags

This page has two versions to it, which can be enabled from the new `feature` tab in the development footer.

Version A: All components are stacked in a single column in the header.  `Get Started` button component only contains a button, no text.

Version B: Same as version A except will have a two-column layout.  `Get Started` button component contains some text.

## QA Notes

There's not a lot of functionality with this component beyond the button and search bar.  The `Get Started` button will take the user to the register page to create an account.  The search bar will take the user to the search page with any query params they searched for.  

Please check that this works on all of our supported browsers and that the breakpoints work in a way that looks and makes sense.

## Ticket

https://openscience.atlassian.net/browse/ENG-91

# Reviewer Checklist
